submission/server/main.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2019 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/google/certificate-transparency-go/submission"
+)
+
+// Flags.
+var (
+	httpEndpoint           = flag.String("http_endpoint", "localhost:5951", "Endpoint for HTTP (host:port)")
+	logListPath            = flag.String("loglist_path", "https://www.gstatic.com/ct/log_list/log_list.json", "Path for list of CT Logs in JSON format")
+	logListRefreshInterval = flag.Duration("loglist_refresh_interval", 2*24*time.Hour, "Interval between consecutive reads of Log-list")
+	rootsRefreshInterval   = flag.Duration("roots_refresh_interval", 24*time.Hour, "Interval between consecutive get-roots calls")
+	policyType             = flag.String("policy_type", "chrome", "CT-policy <chrome|apple>")
+	dryRun                 = flag.Bool("dry_run", false, "Whether stub Log client should be used instead of prod one")
```

Comment:
The "stub Log client" bit is an implementation detail. The description should rather mention what it does, e.g.:
`"Only print CT log requests instead of actually sending them"` (or whatever the stub client does).

Reply:
Rewrote